### PR TITLE
Refactor `ray_julia_jll` into a submodule

### PR DESCRIPTION
**Pull request by omus**
Converts `ray_julia_jll` from a package to a submodule in Ray.jl. See #124 for additional context.

PR #160 should be merged first.

### CODE REVIEW

**Comment by omus** [comment]
The Artifact CI workflows are failing due to:
```
ERROR: LoadError: Some tests did not pass: 267 passed, 0 failed, 10 errored, 0 broken.
in expression starting at /home/runner/work/Ray.jl/Ray.jl/test/runtests.jl:18
UndefVarError: _shutdown_driver not defined
```
This is expected as the binaries contain `shutdown_driver` and not `_shutdown_driver`. I'm inclined to ignore these CI failures as the only reason they were run was because I just moved the Artifacts.toml file.

File: test/ray_julia_jll/runtests.jl [hunk 1]
```diff
@@ -16,10 +16,10 @@ end
     include("address.jl")
     include("objectid.jl")
 
-    setup_ray_head_node() do
+    setup_ray_head_node_basic() do
         # GCS client only needs head node
         include("gcs_client.jl")
-        setup_core_worker() do
+        setup_core_worker_basid() do
```
Comment (glennmoy) [inline test/ray_julia_jll/runtests.jl:22]:
```suggestion
        setup_core_worker_basic() do
```

File: build/wrapper.cc [hunk 2]
```diff
@@ -482,7 +482,7 @@ JLCXX_MODULE define_julia_module(jlcxx::Module& mod)
         .method("Hex", &TaskID::Hex);
 
     mod.method("initialize_driver", &initialize_driver);
-    mod.method("shutdown_driver", &shutdown_driver);
+    mod.method("_shutdown_driver", &shutdown_driver);
```
Comment (glennmoy) [inline build/wrapper.cc:485]: why the underscore? I don't see this being called anywhere?

**Review by glennmoy — APPROVED** [review]
I think we still need to delete the `Ray.jl/ray_julia_jll` directory right?

File: test/ray_julia_jll/runtests.jl (hunk 1 above)
Comment (omus) [reply test/ray_julia_jll/runtests.jl:22]: Thanks. Of course the last minute change has a typo

File: build/wrapper.cc (hunk 2 above)
Comment (omus) [reply build/wrapper.cc:485]: See this commit: https://github.com/beacon-biosignals/Ray.jl/pull/158/commits/4978bf1801b761d7df7987a4c7bf2e28140eb80e

This is being called from `ray_julia_jll.shutdown_driver()`

**Comment by omus** [comment]
There is some additional expected Dockerfile changes to do

**Comment by omus** [comment]
I've fixed the Dockerfile and locally tested it. Once CI has finished I'll merge this PR.

**Pull request merged**